### PR TITLE
Official Version 1 Support

### PR DESCRIPTION
**Pull request by nibanks**
Adds support for the official version 1 of QUIC! Removes support for some of the old draft versions that aren't practically deployed to minimize support costs.

### CODE REVIEW

File: src/core/connection.c [hunk 1]
```diff
@@ -2559,17 +2552,10 @@ QuicConnProcessPeerTransportParameters(
         }
 
         //
-        // Version draft-28 and later fully validate all exchanged connection IDs.
-        // Version draft-27 only validates in the Retry scenario.
+        // Fully validate all exchanged connection IDs.
         //
-        if (Connection->Stats.QuicVersion == QUIC_VERSION_DRAFT_27) {
-            if (!QuicConnValidateTransportParameterDraft27CIDs(Connection)) {
```
Comment (nibanks) [inline src/core/connection.c:2566]: Remove this function too

File: src/core/binding.c [hunk 2]
```diff
@@ -1432,12 +1432,8 @@ QuicBindingDeliverDatagrams(
         // Only Initial (version specific) packets are processed from here on.
         //
         switch (Packet->Invariant->LONG_HDR.Version) {
-        case QUIC_VERSION_DRAFT_27:
```
Comment (ThadHouse) [inline src/core/binding.c:1435]: http didn't depend on any of these versions, right? We won't have to make changes in there by removing these?

Comment (nibanks) [reply src/core/binding.c:1435]: Correct. They have no QUIC version specific code.

**Comment by ThadHouse** [comment]
This probably isn't a blocker, but all the Stub handshake tests are failing. Other then that lgtm.

**Review by ThadHouse — APPROVED** [review]

**Comment by nibanks** [comment]
> This probably isn't a blocker, but all the Stub handshake tests are failing. Other then that lgtm.

I'm looking into them. I think there was always a bug, but the change in TLS code point is now triggering it. I'll try to fix before I merge. Thanks for the review!

**Review by ThadHouse — APPROVED** [review]

File: src/core/crypto_tls.c [hunk 3]
```diff
@@ -387,7 +386,21 @@ QuicCryptoTlsReadExtensions(
                 return Status;
             }
 
-        } else if (ExtType == TlsExt_QuicTransportParameters) {
+        } else if (
+            Connection->Stats.QuicVersion != QUIC_VERSION_DRAFT_29 &&
```
Comment (anrossi) [inline src/core/crypto_tls.c:390]: we're sure we'll never get into the code path except when the version is 29 or 1?

**Review by anrossi — APPROVED** [review]

**Pull request merged**